scipy/stats/_axis_nan_policy.py

```diff
@@ -555,12 +544,11 @@ def hypotest_fun(x):
                         return result_object(*res)
                     return hypotest_fun_out(*samples, **kwds)
 
-            x = np.moveaxis(x, axis, -1)
-            res = np.apply_along_axis(hypotest_fun, axis=-1, arr=x)
-            if n_outputs > 1:
-                res = result_unpacker(res)
-            return _add_reduced_axes(res, reduced_axes, n_outputs, keepdims,
-                                     result_object)
+            x = np.moveaxis(x, axis, 0)
+            res = np.apply_along_axis(hypotest_fun, axis=0, arr=x)
```

Comment:
This makes the output of `np.apply_along_axis` consistent with that of the natively-vectorized function - the zeroth axis is for the separate outputs - except in the case of functions with only one output, and that's why we call `result_unpacker` next.